Support sym round and ceil

GregoryComer opened the pull request:
Summary: Add prim op implementations for sym round and ceil. Support round in pass_base interpreter.

Differential Revision: D65382714

malfet commented on kernels/prim_ops/register_prim_ops.cpp:
```diff
@@ -303,6 +303,51 @@ static Kernel prim_ops[] = {
+          if (a.isDouble()) {
+            out = EValue(static_cast<int64_t>(ceil(a.toDouble())));
+          } else {
+            ET_CHECK_MSG(false, "%zu", (size_t)a.tag);
```
This feels a bit too succinct, please add some explanation why it's unimportant to have it for other dtypes

malfet commented on kernels/prim_ops/register_prim_ops.cpp:
```diff
@@ -303,6 +303,51 @@ static Kernel prim_ops[] = {
+
+            out = EValue(static_cast<int64_t>(res));
+          } else {
+            ET_CHECK_MSG(false, "%zu", (size_t)a.tag);
```
Same as above, needs better error message

malfet commented on kernels/prim_ops/register_prim_ops.cpp:
```diff
@@ -303,6 +303,51 @@ static Kernel prim_ops[] = {
+            const auto d = r - val;
+            auto res = 0.0;
+
+            if ((d != 0.5) && (d != -0.5)) {
```
Nit
```suggestion
            if (abs(d)  != 0.5) {
```

GregoryComer committed:
Support sym round and ceil (#6699)

Summary:
Add prim op implementations for sym round and ceil. Support round in pass_base interpreter.


Test Plan:
Added tests for ceil and round in prim_ops_test.cpp. Round is also test end to end with MSGR ODNC model.

`buck test executorch/kernels/prim_ops/test:`

Reviewed By: malfet

Differential Revision: D65382714

Pulled By: GregoryComer
